### PR TITLE
feat: add tracing (MAPCO-3914)

### PR DESCRIPTION
**Pull request by nitzanmorr**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Further  information:

Added traces using "telemetry" package new span decorators

### CODE REVIEW

File: helm/templates/configmap.yaml [hunk 1]
```diff
@@ -14,9 +14,9 @@ data:
   RESPONSE_COMPRESSION_ENABLED: {{ .Values.env.responseCompressionEnabled | quote }}
   LOG_LEVEL: {{ .Values.env.logLevel | quote }}
   LOG_PRETTY_PRINT_ENABLED: {{ .Values.env.logPrettyPrintEnabled | quote }}
-  {{ if .Values.env.tracing.enabled }}
+  {{ if $tracing.enabled }}
```
Comment (asafMasa) [inline helm/templates/configmap.yaml:17]: fix as discussed, check metrics

File: helm/values.yaml [hunk 2]
```diff
@@ -87,9 +91,6 @@ env:
   responseCompressionEnabled: true
   requestPayloadLimit: 1mb
   internalMountDir: /layerSources
-  tracing:
-    enabled: false
-    url: http://localhost:55681/v1/trace
```
Comment (asafMasa) [inline helm/values.yaml:92]: fix mapproxyCache:
    grids: World84CRS

File: src/layers/models/layersManager.ts [hunk 3]
```diff
@@ -99,6 +107,7 @@ class LayersManager {
     this.logger.info(`Successfully updated mosaic: '${mosaicName}'`);
   }
 
+  @withSpanV4
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:110]: remove

File: src/layers/models/layersManager.ts [hunk 4]
```diff
@@ -256,6 +267,7 @@ class LayersManager {
     return sourceProvider.getCacheSource(sourcePath);
   }
 
+  @withSpanV4
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:270]: fix prettier

File: src/layers/models/layersManager.ts [hunk 5]
```diff
@@ -264,6 +276,7 @@ class LayersManager {
     }
   }
 
+  @withSpanV4
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:279]: revert

File: src/layers/models/layersManager.ts [hunk 6]
```diff
@@ -278,13 +291,15 @@ class LayersManager {
     this.addNewLayer(layerRequest.name, sourceCacheTitle, jsonDocument);
   }
 
+  @withSpanV4
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:294]: revert

File: src/layers/models/layersManager.ts [hunk 7]
```diff
@@ -303,6 +318,7 @@ class LayersManager {
     this.addNewLayer(layerRequest.name, redisCacheTitle, jsonDocument);
   }
 
+  @withSpanV4
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:321]: revert

File: src/layers/models/layersManager.ts [hunk 8]
```diff
@@ -278,13 +291,15 @@ class LayersManager {
...
   private addNewLayer(layerName: string, sourceCacheTitle: string, jsonDocument: IMapProxyJsonDocument): void {
     this.logger.info(`adding ${layerName} to layer list`);
 
     const newLayer: IMapProxyLayer = this.getLayerValues(layerName, sourceCacheTitle);
     jsonDocument.layers.push(newLayer);
   }
 
+  @withSpanV4
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:302]: revert

File: src/layers/models/layersManager.ts (hunk 4 above)
Comment (asafMasa) [reply src/layers/models/layersManager.ts:270]: revert

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: helm/templates/configmap.yaml (hunk 1 above)
Comment (nitzanmorr) [reply helm/templates/configmap.yaml:18]: done

File: helm/values.yaml (hunk 2 above)
Comment (nitzanmorr) [reply helm/values.yaml:92]: fixed

**Pull request merged**